### DISCUSSION TITLE
Allow custom Route export field

### DISCUSSION THREAD

**Discussion by theo-lubert** [vercel/next.js#55140] [Ideas]
## Summary

I'm trying to add a `export const customConfig = ...`  to my `route.ts` (App Router), but Next.js only allows `GET`, `POST`, `PUT`, etc  
```sh
Type error: Route "app/(server)/api/test/route.ts" does not match the required types of a Next.js Route.
  "customConfig" is not a valid Route export field.
```

That limitation forces me to have a separate file, that implement everything in exactly the same way, which I then export in `route.ts` like so:  
```ts
export { OPTIONS, HEAD, GET, POST, PUT, PATCH, DELETE } from './server'
```

Which is... annoying I'd say  
And I understands why this doesn't work, this not a bug, and it is even documented [here (file-conventions/route)](https://nextjs.org/docs/app/api-reference/file-conventions/route), but I ask myself: why is it this strict ?

Basically what I'm wondering about is:
- Why do we have this limitation ?
- Is there any way to get around this behavior ? (without disabling Next.js TS plugin entirely)
- Would a PR be welcome to allow this entirely ? (Draft: https://github.com/vercel/next.js/pull/55157)
- or at least one that make it less strict by allowing prefixed kerys (like `_test` for exemple) ?

Any info on this ?

#### Source
I found:
-  the error message in the code [here (typescript/diagnosticFormatter.ts#L89)](https://github.com/vercel/next.js/blob/12252c271140657164f34290a35a07f0a2d8d279/packages/next/src/lib/typescript/diagnosticFormatter.ts#L89)
- the actual type definition [here (webpack/plugins/next-types-plugin/index.ts#L57)](https://github.com/theo-lubert/next.js/blob/12252c271140657164f34290a35a07f0a2d8d279/packages/next/src/build/webpack/plugins/next-types-plugin/index.ts#L57)

**Comment by theo-lubert** [comment]
Draft PR: https://github.com/vercel/next.js/pull/55157

**Comment by theo-lubert** [comment]
I changed the category to "Ideas" as I actualy propose a change, with an available PR

**Reply by allestaire** [reply]
I guess they did not approved the PR, tsk tsk,
Its so confusing why they need to limit ttsk tsk

**Reply by theo-lubert** [reply]
@allestaire  It did not get much attention, and has not been looked at by the team yet.

As of why there is this limitation: I think the goal is to improves the DX by eliminating all possible typos, which is fair. Yet, the `_` pattern is used in other parts of Next.js to allow this kind of things, so I think it's still relevant 

The fix is actually very simple (PR arleady available), and doesn't interfere with the intended behaviour/goal.  
Based on file ownership, maybe one of @timneutkens @ijjk @shuding @huozhi @ztanner @feedthejim could greenlight this ?

**Reply by theo-lubert** [reply]
I moved the PR from `Draft` to `Open`, we'll if it helps or not

**Reply by allestaire** [reply]
Ahh I see,  thanks for the response 😄
